Src/java/cql-to-elm/src/main/java/org/cqframework/cql/cql2elm/Cql2ElmVisitor.java

```diff
@@ -426,28 +616,55 @@ else if (libraryBuilder.getNamespaceInfo() != null) {
     }
 
     @Override
+    @SuppressWarnings("unchecked")
     public UsingDef visitUsingDefinition(@NotNull cqlParser.UsingDefinitionContext ctx) {
-        Model model = getModel(parseString(ctx.modelIdentifier()), parseString(ctx.versionSpecifier()));
-        return libraryBuilder.resolveUsingRef(model.getModelInfo().getName());
+        List<String> identifiers = (List<String>)visit(ctx.qualifiedIdentifier());
+        String unqualifiedIdentifier = identifiers.remove(identifiers.size() - 1);
+        String namespaceName = identifiers.size() > 0 ? String.join(".", identifiers) :
+                libraryBuilder.getWellKnownNamespaceName(unqualifiedIdentifier);
+
+        String path = null;
+        NamespaceInfo modelNamespace = null;
+        if (namespaceName != null) {
+            String namespaceUri = libraryBuilder.resolveNamespaceUri(namespaceName, true);
+            path = NamespaceManager.getPath(namespaceUri, unqualifiedIdentifier);
+            modelNamespace = new NamespaceInfo(namespaceName, namespaceUri);
+        }
+        else {
+            path = unqualifiedIdentifier;
+        }
+
+        String localIdentifier = ctx.localIdentifier() == null ? unqualifiedIdentifier : parseString(ctx.localIdentifier());
+        if (!localIdentifier.equals(unqualifiedIdentifier)) {
+            throw new IllegalArgumentException(
+                    String.format("Local identifiers for models must be the same as the name of the model in this release of the translator (Model %s, Called %s)",
```

Comment:
Is this a new requirement added to the spec or specific to this implementation?

Reply:
Specific to this implementation. This is mostly a safety net because there are tons of places that make the assumption they're the same and I wasn't comfortable allowing the variation. Probably worth a separate issue to track.

Reply:
Tracking in #611